cjs/interface/event-target.js

```diff
@@ -15,24 +48,37 @@ class DOMEventTarget extends EventTarget {
     return null;
   }
 
+  addEventListener(type, listener, options) {
+    const secret = wm.get(this);
+    const listeners = secret[type] || (secret[type] = []);
+    if (listeners.some(info => info.listener === listener)) {
+      return;
+    }
+    listeners.push({target: this, listener, options});
+  }
+
+  removeEventListener(type, listener) {
+    const secret = wm.get(this);
+    const listeners = secret[type] || (secret[type] = []);
+    secret[type] = listeners.filter(info => info.listener !== listener);
```

Comment:
not sure if we want to do that or keep the `splice` to actually update the initial array. I chose to update it for the sake of the discussion.